kuma/wiki/forms.py

```diff
@@ -295,6 +305,34 @@ def clean_slug(self):
 
         return slug
 
+    def clean_tags(self):
+        """
+        Validate the tags ensuring we have no case-sensitive duplicates.
+        """
+        tags = self.cleaned_data['tags']
+        dupe_tags = []
+
+        if tags:
+            for tag in parse_tags(tags):
+                # Note: The exact match query doesn't work correctly with
+                # MySQL with regards to case-sensitivity. If we move to
+                # Postgresql in the future this code may need to change.
+                doc_tag = DocumentTag.objects.filter(
+                    name__exact=tag).values_list('name', flat=True)
+                # Write a log we can grep to help find duplicate document tags.
+                if len(doc_tag) > 1:
+                    log.warn('Found duplicate document tags: %s' % doc_tag)
+
+                if doc_tag:
```

Comment:
The thing is, I want them all for the logging. Then only the first one (arbitrarily) as the one I'm comparing against. To make it less arbitrary I probably should order by ID and pick the one with the lowest ID as it is likely the one that came first. In the case of `['JavaScript', 'Javascript']` that's true -- `'JavaScript'` is the one we want.